src/Tasks/AssemblyDependency/Reference.cs

```diff
@@ -1042,6 +1042,38 @@ internal FrameworkName FrameworkNameAttribute
             }
         }
 
+        /// <summary>
+        /// Indicates whether <see cref="ResolveAssemblyReference" /> should search for dependencies: assembly references,
+        /// and more esoterically, external files of multi-file assemblies (a.k.a "scatter files" in MSBuild).
+        ///
+        /// Defaults to true, but FindDependencies != true metadata can be set on Reference items to override. This is
+        /// used by the targets that raise NuGet PackageReference items to Reference items as an optimization.
+        /// </summary>
+        internal bool FindDependencies
+        {
+            get
+            {
+                string findDependencies = _primarySourceItem?.GetMetadata("FindDependencies");
+                return string.IsNullOrEmpty(findDependencies) ? true : findDependencies.Equals("true", StringComparison.OrdinalIgnoreCase);
```

Comment:
Can you please cache this value? It traverses the metadata name, news up a StringBuilder and other small things. Probably not a big gain though.